controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaInterceptor.java

```diff
@@ -49,12 +49,14 @@
     public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
 
         Map<String, String> paramMap = new HashMap<>();
-        headers.keys().stream().forEach(key -> {
+        headers.keys().stream()
+               .filter(key -> !key.endsWith(Metadata.BINARY_HEADER_SUFFIX))
```

Comment:
Could you explain why you are adding this filter? What is it filtering exactly and how does it help with not polluting the logs?

Reply:
I believe that this is related to the design of the auth plugin, which allows for arbitrary headers to be passed from client to server.  This filter imposes a constraint on the set of allowed headers.

I presented a more constrained and standards-based approach based on the `Authorization` header in #2385.

Reply:
> I believe that this is related to the design of the auth plugin, which allows for arbitrary headers to be passed from client to server. This filter imposes a constraint on the set of allowed headers.

This is part of grpc header processing. This does not impose any new constraint on the set of allowed headers. Earlier we were trying to process all the headers but we expect only string headers. So we can ignore all the other headers safely.

Reply:
Sorry, I mean that it is filtering out certain headers rather than passing all headers.